src/components/common/ColonyCreationWizard/TokenSelector.tsx

```diff
@@ -0,0 +1,212 @@
+  };
+};
+
+const TokenSelector = ({
+  handleComplete,
+  setLoading,
+  setError,
+  extra,
+  label,
+  appearance,
+  addressField = 'tokenAddress',
+  disabled = false,
+}: Props) => {
+  const { formatMessage } = useIntl();
+  const { validateField } = useFormikContext();
+
+  const labelText =
+    label && typeof label === 'object' ? formatMessage(label) : label;
+
+  const [{ value: tokenAddress }] = useField<string>(addressField);
+
+  const [
+    getToken,
+    {
+      data: tokenQuery,
+      loading: isFetchingAddress,
+      error: fetchingTokenError,
+      called: wasQueryCalled,
+    },
+  ] = useLazyQuery<GetTokenByAddressQuery, GetTokenByAddressQueryVariables>(
+    gql(getTokenByAddressDocument),
+  );
+
+  const tokenData = tokenQuery?.getTokenByAddress?.items;
+  const { name, symbol } = tokenData?.[0] || {};
+  const prevTokenAddress = usePrevious(tokenAddress);
+
+  const fetchToken = (event: React.ChangeEvent<any>) => {
+    const inputtedAddress = event.target.value;
+    /*
+     * Guard against updates that don't include a new, valid `tokenAddress`,
+     * or if the form is submitting or loading.
+     */
+    const isTokenAddressValid =
+      inputtedAddress !== prevTokenAddress &&
+      !isFetchingAddress &&
+      inputtedAddress &&
+      isAddress(inputtedAddress);
+
+    if (!isTokenAddressValid) {
+      return;
+    }
+    setError?.(false);
+    getToken({
+      variables: {
+        id: inputtedAddress,
+      },
+    });
+  };
+
+  const isFetchingError = !!fetchingTokenError;
+  const isInputError =
+    !isFetchingAddress &&
+    isAddress(tokenAddress) &&
+    !tokenData?.length &&
+    wasQueryCalled;
+
+  const hasError = isInputError || isFetchingError;
+
+  /*
+   * Use effect required here to keep parent in sync with TokenSelector
+   * and to avoid setting parent state while rendering TokenSelector
+   */
+  useEffect(() => {
+    setLoading?.(isFetchingAddress);
+    setError?.(hasError);
+    if (tokenQuery) {
+      handleComplete?.(tokenQuery);
+    }
+    /*
+     * If handleComplete includes some dependency on the FormikHelpers (e.g. setFieldValue),
+     * it will be recreated on each render, resulting in an infinite loop.
+     */
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isFetchingAddress, hasError, tokenQuery, setError, setLoading]);
+
+  return (
+    /**
+     * @todo Define custom input component for token addresses
+     */
+    <div className={styles.inputWrapper}>
+      <Input
+        name={addressField}
+        label={labelText || MSG.label}
+        extra={extra}
+        {...getStatusText({
+          addressFieldInput: tokenAddress,
+          isFetchingAddress,
+          hasError,
+          tokenData: { name, symbol },
+        })}
+        onChange={(e) => {
+          fetchToken(e);
+          // Timeout ensures pesky formik validates with correct state
+          setTimeout(() => validateField(addressField), 0);
```

Comment:
Do we still need this hack ?

Reply:
I think so unfortunately. (referenced in comment above)